### PR TITLE
SNO-132 Upgrade python related deps

### PR DESCRIPTION
**Pull request by caseylitton**
Look at commits, there are a couple different updates

This goes with encd pr

ENCD-3983 Update to ubuntu 18 / SNO-132 Update to python3.6 #3109
https://github.com/ENCODE-DCC/encoded/pull/3109

### CODE REVIEW

File: base.ini [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [app:app]
 use = egg:snovault#snowflakes
 sqlalchemy.url = postgresql:///snowflakes
-tm.attempts = 3
+retry.attempts = 3
```
Comment (hitz) [inline base.ini]: what is this parameter for?

File: src/snovault/nginx-dev.conf [hunk 2]
```diff
@@ -11,6 +11,12 @@ http {
         server 127.0.0.1:6543;
```
Comment (hitz) [inline src/snovault/nginx-dev.conf]: This should probably be removed from snovault.

File: src/snovault/tests/elasticsearch_fixture.py [hunk 3]
```diff
@@ -19,7 +19,7 @@ def server_process(datadir, host='127.0.0.1', port=9201, prefix='', echo=False):
         print('IN TRAVIS')
```
Comment (hitz) [inline src/snovault/tests/elasticsearch_fixture.py]: Travis is not used, no longer relevant claus

File: src/snovault/tests/elasticsearch_fixture.py [hunk 4]
```diff
@@ -19,7 +19,7 @@ def server_process(datadir, host='127.0.0.1', port=9201, prefix='', echo=False):
...
         echo = True
         args.append('-Epath.conf=%s/conf' % os.environ['TRAVIS_BUILD_DIR'])
-    else:
+    elif os.path.exists('/etc/elasticsearch'):
         print('NOT IN TRAVIS')
         args.append('-Epath.conf=./conf')
```
Comment (hitz) [inline src/snovault/tests/elasticsearch_fixture.py]: what happens if /etc/elasticsearch doesn't exist?

File: src/snovault/tests/test_snowflake_hash.py [hunk 5]
```diff
@@ -10,4 +10,4 @@
 @pytest.mark.parametrize(('password', 'pwhash'), TEST_HASHES.items())
```
Comment (hitz) [inline src/snovault/tests/test_snowflake_hash.py]: I feel like we don't need this at all any longer.

File: src/snovault/tests/testing_views.py [hunk 6]
```diff
@@ -417,12 +417,12 @@ def testing_retry(context, request):
     from transaction.interfaces import TransientError
 
     model = context.model
-    request._attempt = getattr(request, '_attempt', 0) + 1
+    attempt = request.environ.get('retry.attempt')
```
Comment (hitz) [inline src/snovault/tests/testing_views.py]: oh this must  be the replacement for tm.attempts.

File: src/snowflakes/tests/test_access_key.py [hunk 7]
```diff
@@ -137,4 +137,4 @@ def test_access_key_uses_snowflake_hash(collections, access_key):
     from snovault.snowflake_hash import SNOWHash
```
Comment (hitz) [inline src/snowflakes/tests/test_access_key.py]: again I don't think this hash encryption has been used for years

File: src/snowflakes/types/access_key.py [hunk 8]
```diff
@@ -87,14 +86,14 @@ def access_key_add(context, request):
...
-            for principal in effective_principals(request)
+            for principal in request.effective_principals
             if principal.startswith('userid.')
         ]
 
     password = None
     if 'secret_access_key_hash' not in request.validated:
         password = generate_password()
-        request.validated['secret_access_key_hash'] = crypt_context.encrypt(password)
+        request.validated['secret_access_key_hash'] = crypt_context.hash(password)
```
Comment (hitz) [inline src/snowflakes/types/access_key.py]: this one IS used though

File: src/snovault/tests/testing_views.py (hunk 6 above)
Comment (caseylitton) [reply src/snovault/tests/testing_views.py]: this is spelled wrong, 

```'retry.attempt'```
vs
```retry.attempts = 3```
in the config.  Which is an update in encoded too.

ya know what's weird, the region indexer issue says it's failing do to max retrys.  Wonder if this is related.

Comment (caseylitton) [reply src/snovault/tests/testing_views.py]: Well, it only shows up in test code though.

File: base.ini (hunk 1 above)
Comment (caseylitton) [reply base.ini]: I've only seen it in tests, we have a test making sure it's 3.

**Pull request merged**